src/SoundPlayer.js

```diff
@@ -245,7 +245,8 @@ class SoundPlayer extends EventEmitter {
         taken.finished().then(() => taken.dispose());
 
         taken.volumeEffect.set(0);
-        taken.outputNode.stop(this.audioEngine.audioContext.currentTime + this.audioEngine.DECAY_TIME);
+        const {audioContext, DECAY_TIME, DECAY_SOON} = this.audioEngine;
+        taken.outputNode.stop(audioContext.currentTime + DECAY_SOON + DECAY_TIME);
```

Comment:
seeing a lot of this makes me thing we should just make an API like - also possibly rename the DECAY_ params to match new terminology / just make them a const here?

```js
getDecayTiming() {
  const {DECAY_TIME: duration, DECAY_SOON: preDecay, currentTime} = this;
  return {
    currentTime,
    preDecay,
    startDecay: currentTime + preDecay,
    duration,
    endDecay: currentTime + preDecay + duration,
    timeConstant: duration / 3
  };
}
```

Reply:
Hmm. I not sure about the first part but on the later I think `DECAY_` should stay on AudioEngine. I think `DECAY_SOON` is a candidate to represent `audioContext.baseLatency` or a default like `0.05`.

Reply:
I think we need better than `_SOON` and `_TIME` now that there are multiple aspects to "DECAY_TIMING" - neither one is particularly descriptive of it's purpose 

- Perhaps `DECAY_WAIT` and `DECAY_DURATION` make more sense?

What do you have against the `getDecayTiming()` I feel like it simplifies most other lines in this PR to some form of `const { startDecay, duration } = engine.getDecayTiming()` - and this isn't like a super-hot path right?

Reply:
I think `getDecayTiming` as written above is trying to do too much. Its trying a little of everything for everyone and makes it confusing how to use each returned member. I think the base idea of returning an object to be destructed is fine, but that it should be more targeted. Different getters for different contexts.